### ISSUE TITLE
Create Contact via API - Gender field / API documentation

### ISSUE THREAD

**Issue by LukasRos** [monicahq/monica#1951]
I'm trying to create a contact via the API on the hosted version at app.monicahq.com, following the documentation here: https://www.monicahq.com/api/contacts

According to the docs, I can send `"gender" : "male"` to _POST /contacts_ but the API expects a _gender_id_ instead. Since there's no API to retrieve available gender IDs I don't know any valid value for this, so I cannot create a contact through the API.

Also, the API expects _is_starred_, which is not documented.

Also, the error messages for missing _gender_id_ and _is_starred_ are in Arabic.

**Comment by mattdp** [comment]
Seconding this issue - here was my offline writeup before seeing there was a bug already, with a bit more testing. 

**Is there a planned fix here? It makes adding a new contact through the API impossible, which is very important.**

--

When writing code to import contacts via the API, I ran into some
issues. Documentation of API is here: https://www.monicahq.com/api/contacts

Here's the fields I'm POSTing to https://app.monicahq.com/api/contacts:

```
{"first_met_date_is_year_unknown"=>true,
 "deceased_date_is_year_unknown"=>true,
 "is_partial"=>0,
 "is_dead"=>0,
 "is_starred"=>0,
 "first_name"=>"Jill",
 "last_name"=>"Redacted",
 "company"=>nil,
 "job"=>"Division Director of redacted",
 "gender"=>"female",
 "linkedin_profile_url"=>"https://www.linkedin.com/in/redacted/"}
```

The main issue is that I'm getting back:

```
response.body
=> "{\"error\":{\"message\":[\"The gender id field is required.\"],\"error_code\":32}}"
```

This is when I try strings like "female", strings like "1", and ints
like 1. According to the docs, strings like "female" should be OK, and
there's no mention of a "gender_id" field.

Additionally, is_starred was required to get rid of a similar error message,
even though the docs make no mention of it. This leads me to believe
the API docs are out of sync with the current implementation. 

Thanks for the great work, and happy 2019 - I'm really looking forward
to getting importation working! Also, it's not pretty, but once it's
working I'd be happy to share my example import code in Ruby if you
think other users might gain from it. It's a decent example of what
you have to do to import a bunch of contacts from another system.

**Comment by mattdp** [comment]
@djaiss One task that should be under five minutes that would unblock us would be letting us know what the valid inputs for gender_id are—that way we can just hardcode it and use the API until the longer-term fix is in place. Would it be possible to post what range of inputs gender_id accepts please?

**Comment by djaiss** [comment]
Thanks for pinging me. I forgot about this issue. I'll do it tonight.

**Comment by djaiss** [comment]
This will be addressed in this PR #2259

Basically, you now have access to new methods in the API to manage genders (CRUD).
That means, you can get a list of all genders in your account, create/update or delete at will. I'll also add documentation for all those methods in the next hour.

**Comment by djaiss** [comment]
The API doc is here: https://www.monicahq.com/api/genders

**Comment by djaiss** [comment]
Ping me here or in a new issue if you need something else! I hope it'll solve your needs.

**Comment by mattdp** [comment]
Amazing response time - thanks Régis!
